### DISCUSSION TITLE
Requirement list

### DISCUSSION THREAD

**Discussion by D0nnaz** [Sensinki/Blok-Tech-Team-4A#33] [General]
Kan iedereen z'n requirements list hier sturen :)

**Comment by D0nnaz** [comment]
### Must Have:
* Tijdstempels: De chat de tijd dat het bericht gestuurd is weergeven bij elk bericht, zodat de gebruiker weet hoe laat iets gestuurd is.
* Inlogfunctie: Gebruikers moeten kunnen inloggen met een account, zodat hun chatgeschiedenis en profielinformatie behouden blijven.
* Zichtbaarheid van gebruikersnaam in de chat: De gebruikersnaam van elke deelnemer moet zichtbaar zijn bij hun berichten in de chat, zodat de communicatie duidelijk en gepersonaliseerd is.
* Groepschats voor verschillende talen: De app moet de mogelijkheid bieden om in verschillende groepschats te praten met andere gebruikers.
* Realtime communicatie: De chat moet realtime communicatie ondersteunen, waarbij berichten direct worden verzonden en ontvangen zonder vertraging.
* Nieuwe berichten indicatie: Gebruikers moeten een visuele indicatie krijgen wanneer er nieuwe berichten zijn in een chat, zodat ze geen berichten missen.
* Fallback voor het ontbreken van JavaScript: Als JavaScript niet beschikbaar is, moet de app een fallback-mechanisme hebben om de functionaliteit te behouden of beperkte functionaliteit te bieden zonder afhankelijk te zijn van JavaScript.
* Werken zonder CSS: De app moet goed functioneren, zelfs als CSS-styling niet beschikbaar is, zodat de basisfunctionaliteit behouden blijft.
* Chatgeschiedenis bekijken: De chatgeschiedenis moet zichtbaar zijn, zodat gebruikers eerdere berichten kunnen bekijken.

### Should Have:
* Emojis: De app moet ondersteuning bieden voor emojis, zodat gebruikers emoties kunnen uitdrukken en de communicatie levendiger wordt.
* Gifjes: Het zou handig zijn als de app de mogelijkheid heeft om gifjes te delen, zodat gebruikers visuele elementen kunnen gebruiken om hun boodschap over te brengen.
* Pushmeldingen: Gebruikers moeten pushmeldingen ontvangen wanneer er nieuwe berichten in een chat zijn, zelfs wanneer de app niet actief is, zodat ze direct op de hoogte worden gebracht van nieuwe communicatie.
* Light en dark mode: De app moet een optie bieden om tussen een lichte en donkere modus te schakelen, zodat gebruikers kunnen kiezen welke modus het beste bij hun voorkeuren en omgeving past.
* Bestanden delen tijdens het chatten: Gebruikers moeten de mogelijkheid hebben om bestanden, zoals afbeeldingen of documenten, te delen tijdens het chatten om het leerproces te ondersteunen en te verrijken.
### Could Have:
* Gamification-elementen: De app kan een taalquiz of beloningssysteem bevatten om gebruikers te motiveren hun taalvaardigheden te verbeteren en een element van plezier toe te voegen aan het leerproces.
* Koppeling met social media-platforms: Het zou handig zijn als de app integratie biedt met social media-platforms, zodat gebruikers hun taalpartners beter kunnen leren kennen en hun communicatie kunnen uitbreiden naar andere platforms.
* Matchen van taalpartners: De app kan een mechanisme bevatten om taalpartners te matchen op basis van taalniveau, interesses en andere criteria, zodat gebruikers geschikte gesprekspartners kunnen vinden.
* Gebruikersprofielen: Gebruikers moeten profielen kunnen aanmaken waarin ze hun taalvaardigheid, interesses en andere relevante informatie kunnen weergeven, zodat ze kunnen beslissen met wie ze willen communiceren.
* Zoekfunctie voor taalpartners: Er moet een zoekfunctie zijn waarmee gebruikers specifieke taalpartners kunnen vinden op basis van criteria zoals taal, locatie en interesses.
* Beschikbaarheid en tijdstip: Gebruikers moeten hun beschikbaarheid en gewenst tijdstip voor communicatie kunnen aangeven, zodat er op een handig moment kan worden gecommuniceerd.
* Vertaalfunctie: De app kan een ingebouwde vertaalfunctie hebben, zodat gebruikers chatberichten kunnen begrijpen, zelfs als ze de taal nog niet volledig beheersen.
* Feedbackmechanisme: Gebruikers moeten feedback kunnen geven over hun taalpartners, zodat andere gebruikers kunnen zien welke taalpartners betrouwbaar en behulpzaam zijn.
* (Video) call functie: De app kan een functie hebben waarmee gebruikers (video)gesprekken kunnen voeren om de taal te oefenen met hun taalpartners.

**Comment by Sensinki** [comment]
### Must have
Sign up form with username , e-mail and password
Login an account
Logout an account
Delete an account


### Should have
Dark mode

### Could have
Game foto's uit een API
